src/main/java/com/bean00/RequestReader.java

```diff
@@ -3,32 +3,40 @@
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.List;
 
 public class RequestReader {
     private BufferedReader in;
+    private String requestLine;
+    private List<String> headers = new ArrayList<>();
 
     public RequestReader(BufferedReader in) {
         this.in = in;
     }
 
-    public String getRequestLine() throws IOException {
-        return in.readLine();
-    }
+    public void readRequest() throws IOException {
+        requestLine = in.readLine();
+        System.out.println(requestLine);
 
-    public ArrayList<String> getHeaders() throws IOException {
-        ArrayList<String> headers = new ArrayList<>();
         String header;
 
         do {
             header = in.readLine();
+            System.out.println(header);
```

Comment:
This shouldn't be checked in. It looks like debug code, but I'm pretty sure you meant this to be request. Logging requests is a good idea, but it shouldn't be hidden in this class. When I see that requests are logged to the console, I should be able to find that in an obvious place such as `Server::run`.

Reply:
The null pointer bug we identified yesterday should be addressed no later than when you do the 400 story.

Reply:
1st comment: I'll take out the println statements. I'm also looking into using a Java logging framework to help with logging (which I plan on implementing later).